app/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ target_sources(app PRIVATE src/kscan.c)
 target_sources(app PRIVATE src/matrix_transform.c)
 target_sources(app PRIVATE src/hid.c)
 target_sources(app PRIVATE src/sensors.c)
+target_sources(app PRIVATE src/wpm.c)
```

Comment:
For users that don't care about WPM, we should be able to not have this built at all, including an extra listener, etc.

Can we move this *also* behind a new Kconfig setting?

Reply:
I've put it behind a config session but I'm not sure how I can get the tests to add this for compilation.  Could you point me in the right direction?

Reply:
You'd want to enable it in a file like `app/tests/wpm/1-single_keypress/native_posix.conf` and add the `CONFIG_ZMK_WPM=y` there for each test that needs to enable it.

On that note, I'm fine w/ using just one Kconfig flag for both the widget, and the supporting code for now, but longer term we may want two flags one for the core WPM code, another for the widget, so folks can use the core WPM stuff, but leverage it elsehwere (GATT notification? Keycode that sends some hid keycodes to send the current WPM to the host?) for folks that don't have displays.

Reply:
Right, finally got the tests working! I ended up adding another kconfig option to enable just the wpm calc too. If you're happy with the new commits I'll squash them ready for merge